addons/common/fnc_isTurnedOut.sqf

```diff
@@ -6,75 +6,80 @@ Description:
+    "true" for turned out or "false" for not turned out <BOOL>
     Cargo in exposed vehicles are "turned out" if they lack a
     cargoSoundAttenuation matching index. Do NOT assume that because
     this function returns false that the unit is generally exposed.
 
-    For example, ATVs will return false, even though they are "turned
-    out".
-
 Examples:
     (begin example)
-    if ( [player] call CBA_fnc_isTurnedOut ) then
-    {
-        player sideChat "I am turned out!";
-    };
+        if (player call CBA_fnc_isTurnedOut) then {
+            player sideChat "I am turned out!";
+        };
     (end)
 
 Author:
-    Nou, courtesy of ACRE project.
+    shukari
 ---------------------------------------------------------------------------- */
-SCRIPT(isTurnedOut);
-
 params ["_unit"];
 
-private _turn = false;
-private _out = false;
+if (isNull objectParent _unit) exitWith {true};
+if (isTurnedOut _unit) exitWith {true};
+
+private _config = configFile >> "CfgVehicles" >> typeOf _vehicle;
+private _attenuationType = getText (_config >> "attenuationEffectType");
+if (_attenuationType in ["OpenCarAttenuation", "OpenHeliAttenuation"]) exitWith {true};
+
 private _vehicle = vehicle _unit;
-private _cfg = configFile >> "CfgVehicles" >> (typeOf _vehicle);
-
-if (_vehicle != _unit) then {
-    if (driver _vehicle == _unit) then {
-        private _forceHideDriver = getNumber(_cfg >> "forceHideDriver");
-        _turn = (_forceHideDriver != 1);
-        _out = (_vehicle animationPhase "hatchDriver") > 0;
-    } else {
-        private _assignedRole = assignedVehicleRole _unit;
-        if (_assignedRole select 0 == "Turret") then {
-            private _turretPath = _assignedRole select 1;
-            private _turret = [_vehicle, _turretPath] call CBA_fnc_getTurret;
-
-            private _canHideGunner = getNumber(_turret >> "canHideGunner");
-            private _forceHideGunner = getNumber(_turret >> "forceHideGunner");
-            _turn = (_canHideGunner == 1 && _forceHideGunner == 0);
-
-            private _hatchAnimation = getText(_turret >> "animationSourceHatch");
-            _out = (_vehicle animationPhase _hatchAnimation) > 0;
-
-        } else {
-            if ((_assignedRole select 0) == "Cargo") then {
-                private _attenuateCargo = getArray(_cfg >> "soundAttenuationCargo");
-                if ((count _attenuateCargo) > 0) then {
-                    private _index = (count _attenuateCargo)-1; // wait for command to get cargo index
-
-                    if (_index > -1) then {
-                        if(_index > (count _attenuateCargo)-1) then {
-                            _index = (count _attenuateCargo)-1;
-                        };
-                        if((_attenuateCargo select _index) == 0) then {
-                            _out = true;
-                        };
-                    };
-                };
-            };
+private _role = toLower (_unit call CBA_fnc_vehicleRole);
```

Comment:
`_role` here, cargo index, `_isFFV` and turret path below can all be found from one `fullCrew` call. Something like this:
```
private _fullCrew = fullCrew [_vehicle, ""];
(
    _fullCrew select (_fullCrew findIf {_unit == _x select 0})
) params ["", "_role", "_cargoIndex", "_turretPath", "_isFFV"];
```
I'm not sure if `_cargoIndex` is equal to `getCargoIndex`, worth checking.
Also role should be `toLower`ed then.